overlay/ports/ffmpeg/portfile.cmake

```diff
@@ -40,12 +105,21 @@ if(VCPKG_TARGET_IS_MINGW)
 elseif(VCPKG_TARGET_IS_LINUX)
     string(APPEND OPTIONS " --target-os=linux --enable-pthreads")
 elseif(VCPKG_TARGET_IS_UWP)
-    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-d3d11va --enable-d3d12va --enable-mediafoundation")
+    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-mediafoundation")
+    if(NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
+        string(APPEND OPTIONS " --enable-d3d11va --enable-d3d12va")
+    endif()
 elseif(VCPKG_TARGET_IS_WINDOWS)
-    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-d3d11va --enable-d3d12va --enable-dxva2 --enable-mediafoundation")
+    string(APPEND OPTIONS " --target-os=win32 --enable-w32threads --enable-mediafoundation")
+    if(NOT ("fedora-ffmpeg-free-safe" IN_LIST FEATURES))
+        string(APPEND OPTIONS " --enable-d3d11va --enable-d3d12va --enable-dxva2")
```

Comment:
Direct 3D 11/12 and Direct X are shipped by Microsoft. So they are on the same level as Mediafoundation. 
So I prefer all or nothing.

You have disabled "audiotoolbox" below. If we argue the same here, we need also disable Mediafoundation as non free.

Reply:
I disagree, Direct 3D 11/12 and DirectX are fundamental hardware acceleration APIs also used by Qt etc.
Disabling these APIs solves no known problem, but might slow down the performance and raise the power consumption of Mixxx.

Reply:
Here we are taking about ffmpeg, not Mixxx and Qt. I have no clue how the implications of ffmpeg to these flags. 
Let's check.

Reply:
There is no reason to expand the scope of this PR. And we shouldn't alter the port uneccessary. The patent issue is a valid reason to modify it, the bug in the mediafoundation soundsource too, but everything else should not be altered!

Reply:
It looks like we are talking about hardware accelerated codes listed here: 
https://github.com/FFmpeg/FFmpeg/blob/4ff1d1f3d641cd7a896554e7ac7709193864511d/libavcodec/hwaccels.h#L22

So there are some among them that are free and enabled by fedora. Conclusion We should enable 
`--enable-d3d11va --enable-d3d12va --enable-dxva2` unconditionally. 
The question is, shall we enable "mediafoundation"? 

We may experiment with it to work around the AAC decoder issue on Windows 11.

Reply:
The reason for excluding `d3d11va`, `d3d12va` and `dxva2` is because their sources is not included in Fedora allowlist (see [list](https://src.fedoraproject.org/rpms/ffmpeg/blob/f41/f/ffmpeg_free_sources)), unlike `mediafoundation` which is part of Fedora allowed sources. Happy to disable `mediafoundation` if you think this is the correct course of action here.

Reply:
Of cause d3d11va, d3d12va and dxva2 are not in Fedoras list, as these are APIs of the Windows operating system.

Reply:
The fedora list still includes sources for Android and Media Foundation so I cannot say for sure this is the reason. My fear is that as they would review the sources, those flagged suspicious and they were left out for safety.

If you can guarantee there are 100% patent free, they can be added to the list extension in which I already added file_open.c which looks pretty safe

Reply:
I can gurantee that you can't build these on any other system than Windows, as these APIs are part of Windows. If there are patents, it's Microsoft duty to pay for them, as they ship it as part of their product.
This is the same as with any other Windows or macOS API.

Reply:
I thing we can safely disable mediafoundation because it provides: 
- aac_mf
- ac3_mf 
- av1_mf
- h264_mf 
- hevc_mf
- mp3_mf
All are not enabled. Or is there another hidden dependency? 

It looks like the direct x acceleration is used for the internal video codecs. Mixxx don't uses them either.

Reply:
Right, I see.

Anyway, we are only using FFMPEG for STEM support, the mediafoudation and and CoreAudio will still provide first class support.
The point of this PR was to ensure support across the board to fdk-acc, which is the best AAC codec in FFMPEG, as per their own wiki.

This PR does just this, so I would appreciate we don't scope creep this PR, especially as the release is due for today and Daniel has asked us not to delay it.

If you think we should allow better support, let's take it as part of 2.7. Also, I'm sure you can understand that working off these two VMs is quite painful and takes me a lot of time to get feedback after each tinkering, so please, let's keep that to a minimum, and leave Window or Mac contributors the duty to go above and beyond with the FFMPEG support